scripts/2.TCGA-MLexample-covariates.py

```diff
@@ -80,108 +79,112 @@
 y.value_counts(True)
 
 
-# ## Pre-process data set
-# TODO: currently running PCA on both train and test partitions
+# ## Set aside 10% of the data for testing
 
 # In[10]:
 
-# Pre-process expression data for use later
-n_components = 100
-scaled_expression = StandardScaler().fit_transform(expression)
-pca = PCA(n_components).fit(scaled_expression)
-explained_variance = pca.explained_variance_
-expression_pca = pca.transform(scaled_expression)
-expression_pca = pd.DataFrame(expression_pca)
-expression_pca = expression_pca.set_index(expression.index.values)
-
-
-# In[11]:
-
-print('fraction of variance explained: ' + str(pca.explained_variance_ratio_.sum()))
-
-
-# In[12]:
-
-# Create full feature matrix (expression + covariates)
-X = pd.concat([covariates,expression_pca],axis=1)
+X = pd.concat([covariates, expression], axis=1)
 print('Gene expression matrix shape: {0[0]}, {0[1]}'.format(expression.shape))
-print('Full feature matrix shape: {0[0]}, {0[1]}'.format(X.shape))
-
-
-# ## Set aside 10% of the data for testing
-
-# In[13]:
+print('Covariates matrix shape: {0[0]}, {0[1]}'.format(covariates.shape))
 
 # Typically, this can only be done where the number of mutations is large enough
-train_index, test_index = next(ShuffleSplit(n_splits=2, test_size=0.1, random_state=0).split(y))
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
 
-X_partitions = {
-    'full': {
-        'train': X.ix[train_index], 
-        'test': X.ix[test_index]
-        },
-    'expressions': {
-        'train': expression_pca.ix[train_index], 
-        'test': expression_pca.ix[test_index]
-        },
-    'covariates': {
-        'train': covariates.ix[train_index], 
-        'test': covariates.ix[test_index]
-        }    
-    } 
-
-y_train = y[train_index]
-y_test = y[test_index]
-
-'Size: {:,} features, {:,} training samples, {:,} testing samples'.format(
-    len(X_partitions['full']['train'].columns), 
-    len(X_partitions['full']['train']), 
-    len(X_partitions['full']['test']))
+# Select the feature set for the different models
+N_COVARIATES = covariates.shape[1]
+def select_feature_set_columns(X, feature_set):
+    if feature_set=='expressions': return X[:, N_COVARIATES:]
```

Comment:
> I think that it may get converted to a numpy array in the pipeline

Yes I think you're likely right. If so, just moving it closer to it's usage will be helpful.